lib/ipmi_fru.c

```diff
@@ -4767,6 +4767,46 @@ f_type, uint8_t f_index, char *f_string)
 	}
 	/* Convert index from character to decimal */
 	f_index= f_index - 0x30;
+	
+	if (f_type == 'b' && f_index == 9)		// use index '9' for editing 'board mfg date', by slash.wu@ztsystems.com
+	{
+		const uint64_t secs_from_1970_1996 = 820454400;
+		int y, m, d, h, min;
+		struct tm board_mfg_date;
+		time_t total_secs;
+	
+		sscanf(f_string, "%4d%2d%2d%2d%2d", &y, &m, &d, &h, &min);
+		board_mfg_date.tm_year = y - 1900;
+		board_mfg_date.tm_mon = m - 1;
+		board_mfg_date.tm_mday = d;
+		board_mfg_date.tm_hour = h;
+		board_mfg_date.tm_min = min;
+		board_mfg_date.tm_sec = 0;
+		total_secs = mktime(&board_mfg_date);
+		total_secs -= secs_from_1970_1996;
+		total_secs /= 60;			// mins
```

Comment:
Please create a counterpart for the existing `ipmi_strftime()` (see `ipmi_time.h`). The proposed name is `ipmi_strptime()`. Don't reinvent the bicycle with an `sscanf()` and manual field assignments, use `strptime()` and `mktime()` to get a `time_t` (seconds) value from a string.